File: 3/code3WithJson.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Newtonsoft.Json;

class Program
{
    class Revenue
    {
        public int dia { get; set; }
        public double valor { get; set; }
    }

    static void Main()
    {
        string jsonPath = "data/revenue.json";
        string jsonData = File.ReadAllText(jsonPath);

        List<Revenue> revenues = JsonConvert.DeserializeObject<List<Revenue>>(jsonData);
        var validDays = revenues.Where(r => r.valor > 0).ToList();

        double min = validDays.Min(r => r.valor);
        double max = validDays.Max(r => r.valor);
        double average = validDays.Average(r => r.valor);
        int daysAboveAverage = validDays.Count(r => r.valor > average);

        Console.WriteLine($"Smallest revenue: {min}");
        Console.WriteLine($"Largest revenue: {max}");
        Console.WriteLine($"Days above average: {daysAboveAverage}");
    }
}

File: 3/code3WithJson.ts
import * as fs from "fs";

interface Revenue {
    dia: number;
    valor: number;
}

const data = fs.readFileSync("data/revenue.json", "utf8"); 
const revenues: Revenue[] = JSON.parse(data);

const validDays = revenues.filter((r) => r.valor > 0);
const min = Math.min(...validDays.map((r) => r.valor));
const max = Math.max(...validDays.map((r) => r.valor));
const average = validDays.reduce((sum, r) => sum + r.valor, 0) / validDays.length;
const daysAboveAverage = validDays.filter((r) => r.valor > average).length;

console.log(`Smallest revenue: ${min}`);
console.log(`Largest revenue: ${max}`);
console.log(`Days above average: ${daysAboveAverage}`);
